packages/hdwallet-metamask/src/metamask.ts

```diff
@@ -72,6 +73,16 @@ export class MetaMaskHDWalletInfo implements core.HDWalletInfo, core.ETHWalletIn
     return chainId === 1;
   }
 
+  public async ethGetChainId(): Promise<number> {
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: true, silent: false, timeout: 3000 });
+    if (!provider) {
+      throw new Error("Cannot get chainId");
+    }
+    // chainId as hex string
```

Comment:
Note: Web currently checks for the hex string, but now that we have a proper method for getting the chainId and we're just accessing the `provider.chainId` property, there's probably no reason to keep it as hex string anymore.